ntoskrnl/io/pnpmgr/pnpreport.c

```diff
@@ -464,9 +464,9 @@ IoReportTargetDeviceChange(IN PDEVICE_OBJECT PhysicalDeviceObject,
     ASSERT(notifyStruct->FileObject == NULL);
 
     /* Do not handle system PnP events */
-    if ((RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_QUERY_REMOVE), sizeof(GUID)) != sizeof(GUID)) ||
-        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_CANCELLED), sizeof(GUID)) != sizeof(GUID)) ||
-        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_COMPLETE), sizeof(GUID)) != sizeof(GUID)))
+    if (IsEqualGUID(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_QUERY_REMOVE)) ||
+        IsEqualGUID(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_CANCELLED)) ||
+        IsEqualGUID(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_COMPLETE)))
```

Comment:
The parentheses around the `GUID_....`are redundant, you could remove them. (Same comment also for the other changes below.)
Also, it seems that these comparisons should be `!IsEqualGUID(...)` (i.e. you've introduced a logic error that needs to be fixed). Here and below.

Reply:
>  Also, it seems that these comparisons should be !IsEqualGUID(...) (i.e. you've introduced a logic error that needs to be fixed). Here and below.

Here we want to only handle the custom PnP events instead of the system events, so this is the actual fix.

Reply:
Ah, so you mean that the old code was actually buggy? hehe